lib/node-token-validation/src/api/TokenValidator.ts

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { INetworkModule, Logger } from "@azure/msal-common";
+import { jwtVerify, createRemoteJWKSet, JWTVerifyOptions, JWTPayload } from "jose";
+import crypto from "crypto";
+import { buildConfiguration, Configuration, TokenValidationConfiguration } from "../config/Configuration";
+import { buildTokenValidationParameters, TokenValidationParameters, ValidationParameters } from "../config/TokenValidationParameters";
+import { TokenValidationResponse } from "../response/TokenValidationResponse";
+import { ValidationConfigurationError } from "../error/ValidationConfigurationError";
+import { name, version } from "../packageMetadata";
+import { OpenIdConfigProvider } from "../config/OpenIdConfigProvider";
+import { ValidationError } from "../error/ValidationError";
+
+export class TokenValidator {
+    private config: TokenValidationConfiguration;
+    protected logger: Logger;
+    protected networkInterface: INetworkModule;
+    protected openIdConfigProvider: OpenIdConfigProvider;
+
+    constructor(configuration: Configuration) {
+        this.config = buildConfiguration(configuration);
+        this.logger = new Logger(this.config.system.loggerOptions, name, version);
+        this.networkInterface = this.config.system.networkClient,
+        this.openIdConfigProvider = new OpenIdConfigProvider(this.config.auth.authority, this.networkInterface, this.logger);
+    }
+
+    async validateToken(token: string, options: TokenValidationParameters): Promise<TokenValidationResponse> {
+        this.logger.verbose("validateToken called");
+        
+        if (!token) {
+            throw ValidationConfigurationError.createMissingTokenError();
+        }
+
+        const validationParams: ValidationParameters = await buildTokenValidationParameters(options, this.config);
+        this.logger.verbose("ValidationParams built");
+            
+        const jwks = await this.getJWKS(validationParams);
+
+        const jwtVerifyParams: JWTVerifyOptions = {
+            algorithms: validationParams.validAlgorithms,
+            issuer: validationParams.validIssuers,
+            audience: validationParams.validAudiences,
+            subject: validationParams.subject
+            // Figure out types here
```

Comment:
What is missing here?

Reply:
Ah. I was figuring out whether type validation was needed. Have since amended.